### PR TITLE
feat(stream): add ability to customize legend label

### LINKED ISSUE #323: Customize legend labels in Stream component
Hello, I've read the docs for Stream and Legend but did not find anything regarding how to customize legend labels. 

I've a situation like this:
```javascript
const data = [
  {
    JD: 10500,
    PC: 6700,
	date: '2018-09'
  },
  {
    JD: 7860,
    PC: 5430,
    date: '2018-08'
  }
];

const sellers = [
  {
    id: 'JD',
    name: 'John Dorian'
  },
  {
    id: 'PC',
    name: 'Percival Cox'
  }
];

const keys = sellers.map(s => s.id);
```

There is a way to display the `name` of each seller inside the legend instead of their id apart from mapping the data array and change it to use seller's name as keys instead of their id?

### PR DESCRIPTION
**Pull request by wyze**
Close #323

### CODE REVIEW

**Comment by wyze** [comment]
Here is an example of intended functionality: https://codesandbox.io/s/nivo-forked-ove5e?file=/src/charts/Stream.tsx

File: website/src/data/components/stream/props.js [hunk 1]
```diff
@@ -233,6 +233,18 @@ const props = [
...
         group: 'Dots',
     },
+    {
+        key: 'legendLabel',
+        help: 'Legend label accessor',
+        description: `
+            Define how to access the legend label of each datum,
+            by default, nivo will look for the \`id\` property.
+        `,
+        type: 'string | (datum: RawDatum): string',
```
Comment (plouc) [inline website/src/data/components/stream/props.js:243]: I don't think the type of `datum` is correct as [we just pass the `id`](https://github.com/plouc/nivo/pull/1562/files#diff-44b3698944ddca2af8ad0c22a159233c92c88739e99d018b7a9b9f09f345dfb6R107) if I understand correctly, maybe we could use an inlined type such as `(datum: { id: string }): string`?

File: packages/stream/index.d.ts [hunk 2]
```diff
@@ -55,6 +56,8 @@ declare module '@nivo/stream' {
         offsetType: StackOffset
         curve: AreaCurve
 
+        legendLabel: PropertyAccessor<T, string>
```
Comment (plouc) [inline packages/stream/index.d.ts:59]: I don't think we can use `T` here.

File: packages/stream/index.d.ts [hunk 3]
```diff
@@ -68,15 +71,15 @@ declare module '@nivo/stream' {
         fillOpacity: number
 
         borderWidth: number
-        borderColor: InheritedColorConfig
+        borderColor: InheritedColorConfig<T>
```
Comment (plouc) [inline packages/stream/index.d.ts:74]: The structure you'll get when defining `borderColor` is:
```ts
interface ComputedDatum {
    id: number
    layer: {
        index: number
        x: number
        value: number
        y1: number
        y2: number
    }[] 
    path: string
    color: string
}
```

File: packages/stream/index.d.ts [hunk 4]
```diff
@@ -68,15 +71,15 @@ declare module '@nivo/stream' {
...
 
         enableDots: boolean
         renderDot: StreamDotsItem
         dotPosition: 'start' | 'center' | 'end'
         dotSize: DatumToNumber | number
-        dotColor: InheritedColorConfig
+        dotColor: InheritedColorConfig<T>
```
Comment (plouc) [inline packages/stream/index.d.ts:80]: I think custom accessors for the dots will receive the following structure:

```ts
interface DotDatum {
    key: string
    id: number
    color: string
    index: number
    x: number
    value: number
    y1: number
    y2: number
}
```

Apparently `Datum`, in the definitions file.

It's a combination of [what's computed for the layer points](https://github.com/plouc/nivo/blob/master/packages/stream/src/hooks.js#L91), and some extra properties passed in https://github.com/plouc/nivo/blob/master/packages/stream/src/StreamDots.js#L35.

**Review by plouc — CHANGES_REQUESTED** [review]

**Review by plouc — APPROVED** [review]
LGTM

**Pull request merged**